File: graphs/reconstruct_itinerary.py
class Solution:
    def findItinerary(self, tickets: List[List[str]]) -> List[str]:
      graph = defaultdict(list)
      for src, dst in sorted(tickets, reverse=True):
          graph[src].append(dst)

      route = []

      def visit(airport):
          while graph[airport]:
              visit(graph[airport].pop())
          route.append(airport)

      visit("JFK")
      return route[::-1]

File: graphs/reconstruct_itinerary.rb
def find_itinerary(tickets)
  tickets = tickets.sort.reverse.group_by(&:first)
  route = []
  visit = lambda{  |airport|
    visit[tickets[airport].pop()[1]] while (tickets[airport] || []).any?
    route << airport
  }
  visit["JFK"]
  route.reverse
end